03 Writing Algorithms/12 Universes/06 Futures/99 Examples.html

```diff
@@ -1,21 +1,83 @@
 <p>The following examples demonstrate some common Futures universes.</p>
 
 <h4>Example 1: Contracts Expiring On Quarter End</h4>
-<p>If you run strategies with a quarterly trading cycle, you can filter Future contracts that expire only on quarter ends to reduce the need of rollovers.</p>
+<p>To reduce the need for rolling over, you can filter Future contracts that expire only on quarter ends to save transaction costs. The following example trades a simple EMA cross strategy on SP500 EMini Future contracts that expire on quarter ends.</p>
 <div class="section-example-container">
 	<pre class="csharp">public class QuarterlyFuturesAlgorithm : QCAlgorithm
 {
+    private Future _future;
+    private ExponentialMovingAverage _ema;
+
     public override void Initialize()
     {
-        var future = AddFuture("ES");
-        future.SetFilter(futureFilterUniverse => futureFilterUniverse.ExpirationCycle(new int[] { 3, 6, 9, 12 }));
+        SetStartDate(2024, 1, 1);
+        SetEndDate(2024, 12, 31);
+        SetCash(10000000);
+
+        // Request ES Future data for trading.
+        _future = AddFuture(
+            Futures.Indices.SP500EMini,
+            extendedMarketHours: true,
+            dataNormalizationMode: DataNormalizationMode.BackwardsRatio,
+            dataMappingMode: DataMappingMode.OpenInterest
+        );
+        // To reduce the transaction cost needed for rolling over contracts, we only filter the ones expiring on quarter ends.
+        _future.SetFilter(futureFilterUniverse =&gt; futureFilterUniverse.ExpirationCycle(new int[] { 3, 6, 9, 12 }));
```

Comment:
Since you are trading `Mapped`, the filter doesn't matter. If we replace 3, 6, 9, 12 for 4, 7, 10, 11, the results are the same. In fact, we trade less if we don't use `DataMappingMode.OpenInterest` because the contract with the highest OI can change more than once during the quarter.